core/src/staked_nodes_updater_service.rs

```diff
@@ -19,21 +20,55 @@ pub struct StakedNodesUpdaterService {
     thread_hdl: JoinHandle<()>,
 }
 
+fn firedancer_write_stake(
+    mvcc: &mut Mvcc,
+    staked_nodes: &HashMap<Pubkey, u64>,
+    staked_nodes_overrides: &HashMap<Pubkey, u64>,
+) {
+    let mut values = staked_nodes.iter().filter(|(pubkey, _)| !staked_nodes_overrides.contains_key(pubkey)).collect::<Vec<_>>();
+    values.extend(staked_nodes_overrides.iter());
+    values.sort_by(|a, b| b.cmp(a));
+
+    let mvcc_data = mvcc.begin_write();
+    mvcc_data[0..8].copy_from_slice(&(values.len() as u64).to_le_bytes());
+
+    let max_elements = (mvcc_data.len() - 8) / 40;
+    if values.len() > max_elements {
+        warn!("staked_nodes len {} exceeds max_elements {}", values.len(), max_elements);
+    }
+    for (i, value) in values.iter().enumerate().take(max_elements) {
+        let offset = 8 + i * 40;
+        mvcc_data[offset..offset + 8].copy_from_slice(&value.1.to_le_bytes());
+        mvcc_data[offset + 8..offset + 40].copy_from_slice(&value.0.to_bytes());
```

Comment:
i think `fd_stake_deser` reads the pubkey first, so just making a note to update there too

Reply:
ok!